pkg/xds/translate.go

```diff
@@ -137,26 +309,22 @@ func TranslateRoute(in *routingv2.HTTPRoute,
 			action.RequestMirrorPolicy = &route.RouteAction_RequestMirrorPolicy{Cluster: name(in.Mirror)}
 		}
 
-		if len(in.Route) == 0 { // build default cluster
-			action.ClusterSpecifier = &route.RouteAction_Cluster{Cluster: defaultCluster}
-		} else {
-			weighted := make([]*route.WeightedCluster_ClusterWeight, len(in.Route))
-			for _, dst := range in.Route {
-				weighted = append(weighted, &route.WeightedCluster_ClusterWeight{
-					Name:   name(dst.Destination),
-					Weight: &types.UInt32Value{Value: uint32(dst.Weight)},
-				})
-			}
+		weighted := make([]*route.WeightedCluster_ClusterWeight, len(in.Route))
+		for _, dst := range in.Route {
+			weighted = append(weighted, &route.WeightedCluster_ClusterWeight{
+				Name:   name(dst.Destination),
```

Comment:
You need a check here. if the cluster is an unresolved cluster, you should skip adding the weighted cluster entry. Ideally, if the destination subset does not exist, you should return a nil from the naming function and appropriately skip adding the cluster.

Reply:
Why so? If I have two clusters in weighted cluster section, where one is resolved and another one isn't, what happens?

Reply:
In that case, the entire rule must be skipped.